CMakeLists.txt

```diff
@@ -79,6 +79,8 @@ set (CMAKE_MODULE_PATH
 include (InstallRequiredSystemLibraries)
 include (GNUInstallDirs)
 
+include(MongoC-Warnings)
```

Comment:
I suggest commenting to note the intent of these warnings as compared to the warnings enabled with `ENABLE_MAINTAINER_FLAGS`. My understanding is:
- `MongoC-Warnings` are always enabled. They should be a subset that should never produce warnings.
- `ENABLE_MAINTAINER_FLAGS` are less critical, but of interest to maintainers.